lib/Backend/LowerMDShared.cpp

```diff
@@ -9307,15 +9324,52 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
 
                 //MOVSD dst, src1;
                 Assert(!dst->IsEqual(src1));
-                this->m_lowerer->InsertMove(dst, src1, instr);
+                
+                IR::Opnd * opndNaN = nullptr;
+
+                bool isF32 = dst->IsFloat32();
+
+                if (isF32)
+                {
+                    opndNaN = IR::MemRefOpnd::New((double*)&(Js::JavascriptNumber::k_Nan32), IRType::TyFloat32, this->m_func);
+                }
+                else
+                {
+                    opndNaN = IR::MemRefOpnd::New((double*)&(Js::JavascriptNumber::k_Nan), IRType::TyFloat64, this->m_func);
+                }
+
+                IR::Opnd * src1f64 = src1;
+                IR::Opnd * src2f64 = src2; 
+                if (propagateNaN)
+                {
+                    if (src1->IsFloat32())
+                    {
+                        src1f64 = IR::RegOpnd::New(TyFloat64, m_func);
+                        EmitFloat32ToFloat64(src1f64, src1, instr);
 
+                        src2f64 = IR::RegOpnd::New(TyFloat64, m_func);
+                        EmitFloat32ToFloat64(src2f64, src2, instr);
+                    }
+
+                    IR::Opnd* isSrc1NaN = IsOpndNaN(src1f64, instr);
+                    IR::Opnd* isSrc2NaN = IsOpndNaN(src2f64, instr);
```

Comment:
actually, it looks like we already do this. i think the only change in lowering that we need is for InsertCompareBranch to accept float32

---
In reply to: [70487092](https://github.com/Microsoft/ChakraCore/pull/659#discussion_r70487092) [](ancestors = 70487092)